Standards/scs-xxxx-v1-provider-network-standard.md

```diff
@@ -0,0 +1,200 @@
+Because such networks break the isolation of the cloud environment, they must be created and managed by the CSP and made available to projects only in a controlled manner.
+
+In OpenStack, such CSP-managed networks are called _provider networks_, though providing access to the internet is not their only use case.
+Depending on the configuration, they may impose different usage restrictions and offer different methods of allocating IP addresses.
+
+External access to and from virtualised resources is a fundamental feature of IaaS, and providing this access is a hard requirement for any cloud provider.
+However, OpenStack provider networks offer some design space regarding usage restrictions and IP address allocation.
+This document defines a standardized setup for provider networks that offers external access in a way that is portable across SCS clouds.
+
+### Glossary
+
+The following terms are used throughout this document:
+
+| Term | Meaning |
+|------|---------|
+| CSP | Cloud Service Provider, provider managing the OpenStack infrastructure. |
+| Server | Server object in the OpenStack Compute API, usually represents a virtual machine, though there are also compute drivers for containers and bare metal servers. |
+| Virtual Network | A virtualized network managed by OpenStack, provides connectivity between servers and other network resources, such as virtual routers. |
+| Virtual Router | OpenStack resource that can be used to route and bridge between virtual networks and provide other features such as NAT |
+| Subnet | Subdivision of available IP address space using address prefixes. In OpenStack also an abstraction for controlling IP address allocation in a virtual network. |
+| DHCP | Dynamic Host Configuration Protocol: Used to automatically configure hosts in a network with IP addresses, default routes, and other information such as DNS servers. |
+| Prefix | IP address prefix of a given bit-length N, written _/N_. Divides addresses into a network and a host part, a shorter prefix allows more hosts but takes up more address space. |
+| NAT | Network Address Translation: mapping (usually public) IPv4 addresses onto a different (usually private) address space. May allows multiple hosts to share a public address by multiplexing TCP/UDP ports. |
+| RBAC | Role-based Access Control: A mechanism in the Network API to give projects limited access to resources owned by other projects. Typically used by CSPs to create provider networks. |
+| Shared Network | Virtual network that is shared between projects in a way that allows direct attachment of servers. |
+| External Network | Virtual network that is shared between projects in a way that only allows virtual routers to use it as external gateway. Typically used by CSPs to provide access to networks outside of the cloud environment. |
+| Provider Network | Any shared or external network that is managed by the CSP. |
```

Comment:
OpenStack provider networks have a different property, namely that they're tied to non-virtualised networks. I think that property is relevant enough to include it in this definition.

(Or are we using the term "provider network" in a different sense here?)

Reply:
In many places, including some [Openstack documentation](https://docs.openstack.org/install-guide/launch-instance-networks-provider.html), I found the term provider network to be used (interchangeably with "external network") to mean CSP-managed networks that are tied to non-virtualized networks, because that is usually what they are used for.

I was looking at this from more of an RBAC-perspective though, where any network, regardless of it's `provider:network_type` and `provider:physical_network` can have an `access_as_external` rule, and where ownership and RBAC rules ultimately dictate how a network can be used in a project. In that context, it made more sense to me to define provider networks literally as "networks managed by the provider".
This will still cover the classical use-case of bridging into non-virtualized networks, but also use-cases like a CSP-managed shared network.

It's still a deviation from common Openstack terminology of course, and I can try rewording it to stay closer to that.

Reply:
> In that context, it made more sense to me to define provider networks literally as "networks managed by the provider".

I agree. My main problem with the definition as written above is that the "managed by the CSP" gets overpowered by the "shared or external" attributes, which make me read that definition as if the key point was that the network was "shared or external", not "managed by the CSP".

Maybe: "A network managed by the CSP, typically connected to non-virtualized infrastructure and commonly made external or shared."

Reply:
The network being shared or external is kinda essential though, as that is what makes it a "provider network" to tenants, otherwise it's just invisible to them and might as well be non-existent.

I have changed it now to "A CSP-managed virtual network made available to projects as either shared or external, typically connected to non-virtualized infrastructure."